src/App.tsx

```diff
@@ -15,49 +16,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'alphabet',
+  Lenghth = 'length',
+  None = '',
+}
+
+function sortByType(goods:string[], query:string, isReversed:boolean) {
+  const prepGoods = [...goods];
+
+  if (query) {
+    switch (query) {
+      case SortType.Alphabet:
+        prepGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case SortType.Lenghth:
+        prepGoods.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        break;
+    }
+  }
+
+  if (isReversed) {
+    prepGoods.reverse();
+  }
+
+  return prepGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.None);
+  const [isReversed, setIsReversed] = useState(false);
+  const sortedGoods = sortByType(goodsFromServer, sortType, isReversed);
+
+  const sortByAlphabet = () => {
+    setSortType(SortType.Alphabet);
+  };
+
+  const sortByLength = () => {
+    setSortType(SortType.Lenghth);
+  };
+
+  const reversed = () => {
+    setIsReversed(hasState => !hasState);
+  };
```

Comment:
Those helpers are redundant you can set state directly on click in this case. Helpers are useful for more complicated logic